frontend/view-edit-client/invoices/edit-invoice.component.tsx

```diff
@@ -294,10 +295,14 @@ const EditInvoice = React.forwardRef(function (props: EditInvoiceProps, ref) {
     const lineItems = modifiedInvoice.lineItems
       .concat(newLineItems)
       .map((li) => ({ ...li, rate: Number(li.rate) }));
+    const clientIds = clientRef.current
+      .getClients()
+      // @ts-ignore
+      .map((item) => item.clientId);
     const result = {
       ...modifiedInvoice,
       lineItems,
-      clients: props.client ? [props.client.id] : [],
+      clients: props.client ? clientIds : [],
```

Comment:
Now that we're guaranteed to have an array of clientIds, I think there's no need to sometimes pass the empty array:

```suggestion
      clients: clientIds,
```